Content.Shared/Lock/Lockpick/LockpickSystem.cs

```diff
@@ -0,0 +1,70 @@
+using Content.Shared.Interaction;
+using Content.Shared.DoAfter;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Lock.Lockpick;
+
+public sealed class LockpickSystem : EntitySystem
+{
+    [Dependency] private readonly LockSystem _lockSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<LockpickComponent, AfterInteractEvent>(LockpickUse);
+        SubscribeLocalEvent<LockpickComponent, LockpickingDoAfterEvent>(LockpickDoAfter);
+    }
+
+    //Checks if target entity has a locked lock component
+    //Starts a lockpicking DoAfter
+    private void LockpickUse(Entity<LockpickComponent> ent, ref AfterInteractEvent args)
+    {
+        if (HasComp<LockComponent>(args.Target) && _lockSystem.IsLocked(args.Target.Value))
+        {
+            if (args.CanReach)
```

Comment:
I'd recommend doing guard statements for both these if statements to avoid big indents. Instead of checking if something is true, check if its false and then return. This is how its usually done for these kind of checks!

For example this line would turn into:

```
if (!args.CanReach)
    return;
```
and you wouldn't need to put everything inside the if statement. 

Also, remember de morgans! The inverse of `a && b` is `!a || !b` NOT `!a && !b`.